content/blog/2020-05-dplyr-1-0-0-last-minute-additions/index.Rmarkdown

```diff
@@ -0,0 +1,147 @@
+---
+title: 'dplyr 1.0.0: last minute additions'
+author: Hadley Wickham
+date: '2020-05-06'
+slug: dplyr-1-0-0-last-minute-additions
+categories:
+  - package
+tags:
+  - dplyr
+description: >
+    Learn about two last minute additions to dplyr 1bl.0.0: a chattier 
+    `summarise()` with more options for controlling grouping of output,
+    and new row manipulation functions inspired by SQL.
+photo:
+  url: https://unsplash.com/photos/FfbVFLAVscw
+  author: Malvestida Magazine
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+This post is the latest in a series of post leading up the the dplyr 1.0.0 release on May 15. So far, the series has covered:
+
+-   [Major lifecycle changes](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-is-coming-soon/).
+-   [New `summarise()` features](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-summarise/).
+-   [`select()`, `rename()`, and (new) `relocate()`](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-select-rename-relocate/).
+-   [Working `across()` columns](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-colwise/).
+-   [Working within rows](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-rowwise/).
+-   [The role of the vctrs package](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-and-vctrs/)
+-   [Notes for package developers](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-package-dev/)
+
+Today I wanted to talk about two cool new features that we've added since I started blogging about dplyr 1.0.0: `summarise()` now gives you greater control over how the results and grouped, and we've added a new set of functions for modifying rows.
+
+### Getting the dev version
+
+If you'd like to try out anything you see in this blog post, you can install the development version of dplyr with:
+
+```{r, eval = FALSE}
+devtools::install_github("tidyverse/dplyr")
+```
+
+Note that the development version won't become 1.0.0 until it's released, but at this point, it's very similar to what we'll be submitting to CRAN on May 15.
+
+```{r setup}
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## `summarise()` and grouping
+
+There\'s a common confusion about the result of `summarise()`. How do you think the result of the following code will be grouped?
+
+```{r, message = FALSE}
+homeworld_species <- starwars %>% 
+  group_by(homeworld, species) %>% 
+  summarise(n = n())
+```
+
+You might be surprised to learn that it's grouped by `homeworld`:
+
+```{r}
+head(homeworld_species, 3)
+```
+
+That's because `summarise()` always peels off the last group, based on the logic that this group now occupies a single row so there's no point grouping by it. This behaviour made perfect sense to me at the time I implemented it, but it's been a long standing source of confusion amongst dplyr users (and it doesn't make sense if your summary [returns multiple rows](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-summarise/)).
```

Comment:
```suggestion
That's because `summarise()` always peels off the last group, based on the logic that this group now occupies a single row so there's no point grouping by it. This behaviour made perfect sense to me at the time I implemented it, but it's been a long standing source of confusion among dplyr users (and it doesn't make sense if your summary [returns multiple rows](https://www.tidyverse.org/blog/2020/03/dplyr-1-0-0-summarise/)).
```

Reply:
Oxford Style Manual has been using among since _at least_ 2002 (sorry, that's the earliest version I've unpacked).